### PR TITLE
NAS-110511 / 21.06 / Bug fixes for setting system birthday

### PR DESCRIPTION
**Pull request by sonicaj**
This PR fixes following issues:

1) We were trusting that system time would be the correct one, but that may not be the case if `ntpd` has not synced. Changes have been introduced to make sure that we trust the time returned by ntp request if only the clock has synchronized.

2) If we were unable to determine system time, we were waiting for 24 hours before checking again for system birthday which means that we would be storing the incorrect system birthday even when we could potentially save the correct time if we just checked sooner.

3) Do not unnecessarily have a hook for `interface.post_sync` when setting system birthday is a one time thing. It will already be ensured that we do set it. So if the first pass failed and interfaces were synced, we will be having another pass after 5 minutes and we will catch up with the correct synced time.

### CODE REVIEW

File: src/middlewared/middlewared/plugins/system.py [hunk 1]
```diff
@@ -1618,41 +1621,18 @@ def set_crash_reporting(self):
...
+    while True:
+        birthday = await middleware.call('system.get_synced_clock_time')
+        if birthday:
+            middleware.logger.debug('Updating birthday data')
+            # update db with new birthday
+            settings = await middleware.call('datastore.config', 'system.settings')
+            await middleware.call('datastore.update', 'system.settings', settings['id'], {'stg_birthday': birthday})
+            break
+        else:
+            middleware.logger.debug('Waiting for system clock to be synchronized')
```
Comment (themylogin) [inline src/middlewared/middlewared/plugins/system.py:1634]: Will it flood middleware logs if clock are never synchronized?

**Review by themylogin — CHANGES_REQUESTED** [review]

**Review by themylogin — APPROVED** [review]

File: src/middlewared/middlewared/job.py [hunk 2]
```diff
@@ -183,7 +183,8 @@ def __init__(self, maxlen=1000):
         self.maxlen = maxlen
         self.count = 0
         self.__dict = OrderedDict()
-        shutil.rmtree(LOGS_DIR)
+        if os.path.exists(LOGS_DIR):
```
Comment (yocalebo) [inline src/middlewared/middlewared/job.py:186]: Looks like the most recent commit might have removed the `with contextlib` context manager?

Comment (yocalebo) [reply src/middlewared/middlewared/job.py:186]: Oh wait, looks like this should be part of the changes for this commit?

Comment (sonicaj) [reply src/middlewared/middlewared/job.py:186]: It would catch up as soon as the related PR is merged - a sec ; )

**Review by yocalebo — APPROVED** [review]

**Pull request merged**